### ISSUE TITLE
Zipkin cluster does not honor TLS parameters

### ISSUE THREAD

**Issue by jeffalder** [envoyproxy/envoy#23507]
*Title*: Zipkin cluster does not seem to honor TLS parameters

*Description*:
I have configured the Zipkin cluster to have min and max TLS version 1.3
```
  - name: zipkin
    connect_timeout: 30s
    type: STRICT_DNS
    dns_lookup_family: V4_ONLY
    lb_policy: ROUND_ROBIN
    wait_for_warm_on_init: false
    transport_socket:
      name: envoy.transport_sockets.tls
      typed_config:
        "@type": type.googleapis.com/envoy.extensions.transport_sockets.tls.v3.UpstreamTlsContext
        common_tls_context:
          tls_params:
            tls_minimum_protocol_version: TLSv1_3
            tls_maximum_protocol_version: TLSv1_3
    load_assignment:
      cluster_name: zipkin
      endpoints:
      - lb_endpoints:
        - endpoint:
            address:
              socket_address:
                address: my-host
                port_value: 443
```

`my-host` is in CloudFlare and **does not support TLSv1.0**. Unfortunately, Envoy 1.23.1 cannot negotiate this TLS connection.

Per Wireshark, the record layer is TLSv1.
![not-working-envoy-1.23](https://user-images.githubusercontent.com/49817386/195909982-669829cd-e0ee-4e6c-bf64-2e9c2805821a.png)

When I attempt to hit this host using `mitmproxy` or `curl`, that record layer shows as TLSv1.3
![working-curl](https://user-images.githubusercontent.com/49817386/195910261-c3846fb3-769e-4db6-ac77-9b9020b3d6e3.png)

I don't understand why Envoy is attempting to negotiate a TLSv1.2 connection when the minimum version is supposed to be TLSv1.3.

*Repro steps*:
1. Configure a host that can respond to HTTPS and return 200s
2. Point a zipkin cluster at it
3. Set the min and max TLS versions to TLSv1.3
4. Use `tcpdump` to capture the packets and observe the TLSv1 record layer.

*Admin and Stats Output*:
[envoy.tar.gz](https://github.com/envoyproxy/envoy/files/9788964/envoy.tar.gz)

*Config*:
See https://github.com/jeffalder/envoy-servicename-repro/blob/main/envoy.yaml; other than adding the min/max versions per above, it's pretty much the same.

*Logs*:
[envoy.pcap.zip](https://github.com/envoyproxy/envoy/files/9788967/envoy.pcap.zip)

After much experimentation, this is the best I could get out of the logs.
```
envoy-experiment-envoy-1    | [2022-10-14 17:48:23.499][16][debug][connection] [./source/common/network/connection_impl.h:89] [C22] current connecting state: true
envoy-experiment-envoy-1    | [2022-10-14 17:48:23.500][16][debug][connection] [source/common/network/connection_impl.cc:924] [C22] connecting to 162.247.241.19:443
envoy-experiment-envoy-1    | [2022-10-14 17:48:23.500][16][debug][connection] [source/common/network/connection_impl.cc:943] [C22] connection in progress
envoy-experiment-envoy-1    | [2022-10-14 17:48:23.732][16][debug][connection] [source/common/network/connection_impl.cc:683] [C22] connected
envoy-experiment-envoy-1    | [2022-10-14 17:48:23.972][16][debug][connection] [source/extensions/transport_sockets/tls/ssl_socket.cc:228] [C22] remote address:162.247.241.19:443,TLS error: 268436496:SSL routines:OPENSSL_internal:SSLV3_ALERT_HANDSHAKE_FAILURE 268435610:SSL routines:OPENSSL_internal:HANDSHAKE_FAILURE_ON_CLIENT_HELLO
envoy-experiment-envoy-1    | [2022-10-14 17:48:23.972][16][debug][connection] [source/common/network/connection_impl.cc:250] [C22] closing socket: 0
envoy-experiment-envoy-1    | [2022-10-14 17:48:23.972][16][debug][connection] [source/extensions/transport_sockets/tls/ssl_socket.cc:228] [C22] remote address:162.247.241.19:443,TLS error: 268436496:SSL routines:OPENSSL_internal:SSLV3_ALERT_HANDSHAKE_FAILURE 268435610:SSL routines:OPENSSL_internal:HANDSHAKE_FAILURE_ON_CLIENT_HELLO
envoy-experiment-envoy-1    | [2022-10-14 17:48:24.887][1][debug][connection] [./source/common/network/connection_impl.h:89] [C23] current connecting state: true
envoy-experiment-envoy-1    | [2022-10-14 17:48:24.888][1][debug][connection] [source/common/network/connection_impl.cc:924] [C23] connecting to 172.18.0.2:80
envoy-experiment-envoy-1    | [2022-10-14 17:48:24.888][1][debug][connection] [source/common/network/connection_impl.cc:943] [C23] connection in progress
envoy-experiment-envoy-1    | [2022-10-14 17:48:24.888][1][debug][connection] [source/common/network/connection_impl.cc:683] [C23] connected
```

**Comment by wbpcode** [comment]
It seems that this problem is none of zipkin's business. And it's weird that no certs/verify config in the tls context. (I am not sure if it's acceptable.)

cc @lizan

**Comment by jeffalder** [comment]
@wbpcode Thanks for looking at this! I certainly wouldn't have expected Zipkin to be related; I figured this was well before the tracing code did anything with the actual upstream, but I didn't want to make assumptions about what was or was not relevant.

The answer was surprisingly simple, yet opaque: someone else pointed out that I didn't have SNI configured as part of my upstream context. Once I added the `sni` field matching the hostname I was using, it started working and I got Server Hellos and successful TLS negotation.

So this is addressed.
